File: 861-flipping-an-image/flipping-an-image.cpp
class Solution {
public:
    vector<vector<int>> flipAndInvertImage(vector<vector<int>>& image) {
        vector<vector<int>> ans;
        for(int i = 0; i < image.size(); i++) {
            reverse(image[i].begin(), image[i].end());
            for(int j = 0; j < image[0].size(); j++) {
                image[i][j] = (image[i][j] & 1) == 1 ? 0 : 1;
            }
            ans.push_back(image[i]);
        }
        return ans;
    }
};

File: 861-flipping-an-image/flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        
        for(int[] arr : image){
            reverse(arr);
            invert(arr);
        }
        return image;
    }

    private void reverse(int[] arr) {
        int start = 0;
        int end   = arr.length - 1;

       while(start < end) {
            int temp  = arr[start];
            arr[start] = arr[end];
            arr[end]  = temp;

            start += 1;
            end   -= 1;
       }  
    }

    private void invert(int[] arr) {
        for(int i = 0; i < arr.length; i++) {
            if(arr[i] == 0) arr[i] = 1;
            else arr[i] = 0;
        }
    }
}